core/net/rpl/rpl-icmp6.c

```diff
@@ -1021,12 +1543,20 @@ dao_input(void)
   }
 
   if(RPL_IS_STORING(instance)) {
-    dao_input_storing();
```

Comment:
If I understood your first question, you're asking if I can enable security in RPL with_RPL_SECURITY_ macro without setting the preinstalled key through_RPL_CONF_SECURITY_K_ macro. The answer is yes. If someone forgets to set the key, a default key is set in `rpl-conf.h`, to prevent possible uncorrect behaviours.

Speaking about the second question, I think Contiki assumes that nodes have only one network interface, but I might be wrong. Anyway, in this implementation if security is enabled, nodes drop non-recurring packets.